api/controllers/posts.js

```diff
@@ -0,0 +1,38 @@
+const Post = require('../models/Post');
+
+async function index (req, res) {
+    try {
+        const posts = await Post.all;
+        res.status(200).json(posts)
+    } catch (err) {
+        res.status(500).json({err})
+    }
+}
+
+async function show (req, res) {
+    try {
+        const post = await Post.findById(req.params.id);
+        res.status(200).json(post)
```

Comment:
You did not use the spread operator on the .json(post), it should be `.json(...post)`